modules/sync/syncs/httpsynchronization.cpp

```diff
@@ -97,24 +98,47 @@ void HttpSynchronization::start() {
     }
     _state = State::Syncing;
 
-    if (hasSyncFile()) {
+    if (checkSyncFile()) {
         _state = State::Resolved;
         return;
     }
 
+    if (isRejected()) {
+        return;
+    }
+
     std::string query = fmt::format(
         "?identifier={}&file_version={}&application_version={}",
         _identifier, _version, ApplicationVersion
     );
 
     _syncThread = std::thread(
         [this](const std::string& q) {
-            for (const std::string& url : _syncRepositories) {
-                const bool success = trySyncFromUrl(url + q);
-                if (success) {
-                    createSyncFile();
-                    _state = State::Resolved;
-                    return;
+        for (const std::string& url : _syncRepositories) {
+            //TODO: handle multiple syncRepositories, before it would only create
+            // syncfile from first success and then continue
+            const bool success = trySyncFromUrl(url + q);
+            if (success) {
+                _state = State::Resolved;
+                createSyncFile(success);
+                return;
+            }
+            else {
+                
+                //If it was not successful we should add any files that were potentially
+                //downloaded, so we dont download them again from the new repository
+                _existingSyncedFiles.insert(_existingSyncedFiles.end(),
+                    _newSyncedFiles.begin(), _newSyncedFiles.end());
```

Comment:
```suggestion
                _existingSyncedFiles.insert(
                    _existingSyncedFiles.end(),
                    _newSyncedFiles.begin(),
                    _newSyncedFiles.end()
                );
```

If the parameters to a function don't fit on the same line, they always each get their own line with the closing `);` on its own line as well